src/datasource.js

```diff
@@ -50,17 +52,12 @@ export class GenericDatasource {
 
     return this.q.all(promises).then(richResponses => {
       var result = _.map(richResponses, (richResponse) => {
-        var response = richResponse.response;
-        var datapoints;
-        if (response.data.length != 0) {
-          datapoints = _.map(response.data[0].data, point => [point.value, point.timestamp]);
-        } else {
-          datapoints = [];
-        }
         return {
           refId: richResponse.refId,
           target: richResponse.target,
-          datapoints: datapoints
+          // The javascript's flatMap
+          datapoints: [].concat.apply([], richResponse.response.data.map(d => d.data))
```

Comment:
It did not, but I changed it in the following commits. See here: https://github.com/hawkular/hawkular-grafana-datasource/pull/10/files#diff-eae92c615e02f3d65b7f014cce8b162fR62 . There isn't this 'funny' construct anymore.